src/live_restore/live_restore_fs.c

```diff
@@ -1154,6 +1156,9 @@ __live_restore_fs_open_in_destination(WT_LIVE_RESTORE_FS *lr_fs, WT_SESSION_IMPL
     WT_FILE_HANDLE *fh;
     char *path;
 
+    /* This function is only called for files. Directories are handled separately. */
+    WT_ASSERT(session, lr_fh->file_type != WT_FS_OPEN_FILE_TYPE_DIRECTORY);
```

Comment:
Should this be an ALWAYS?

Reply:
[1fe24c5](https://github.com/wiredtiger/wiredtiger/pull/11379/commits/1fe24c5407d375cef1de5a0536e4ee4aa57eee51)